src/main/java/fi/tkgwf/ruuvi/strategy/impl/DefaultDiscardingWithMotionSensitivityStrategy.java

```diff
@@ -0,0 +1,30 @@
+package fi.tkgwf.ruuvi.strategy.impl;
+
+import fi.tkgwf.ruuvi.bean.RuuviMeasurement;
+import fi.tkgwf.ruuvi.config.Config;
+import fi.tkgwf.ruuvi.strategy.LimitingStrategy;
+
+import java.util.Optional;
+
+/**
+ * Normally discard measurements that are coming in too fast, except when acceleration has been detected -- in that
+ * case the packet is always saved.
+ * <p>
+ * The time limit is defined as {@link Config#getMeasurementUpdateLimit()}.
+ * The acceleration bounds are defined as {@link Config#getDefaultWithMotionSensitivityStrategyLowerBound()}
+ * and {@link Config#getDefaultWithMotionSensitivityStrategyUpperBound()}.
+ * <p>
+ * The limit is applied separately to all the different devices sending data, i.e. per MAC address.
+ */
+public class DefaultDiscardingWithMotionSensitivityStrategy implements LimitingStrategy {
+    private final DiscardUntilEnoughTimeHasElapsedStrategy defaultStrategy = new DiscardUntilEnoughTimeHasElapsedStrategy();
+
+    @Override
+    public Optional<RuuviMeasurement> apply(final RuuviMeasurement measurement) {
+        if (measurement.accelerationTotal < Config.getDefaultWithMotionSensitivityStrategyLowerBound()
+            || measurement.accelerationTotal > Config.getDefaultWithMotionSensitivityStrategyUpperBound()) {
+            return Optional.of(measurement);
+        }
+        return defaultStrategy.apply(measurement);

```

Comment:
Only calling the default strategy here will prevent updating its internal "last updated" map, which will allow the first "non acceleration measurements" after "acceleration events" to pass through even if the defined measurementUpdateLimit haven't passed yet

Reply:
That's actually a feature, but I guess it should be made more official. :D I want it to have an event where the acceleration values are reset afterwards. Otherwise, if you plot the movements, the acceleration stays high even though it actually ended. I was thinking of adding a parameter that defines how many events should be recorded after an acceleration event. It'd be nice to record the context before the event as well... I could store the previous *n* events (n ~ 3) and submit them first when acceleration takes place, but would that mess the timestamps, or does InfluxDb read the timestamps from the measurements already...?

Reply:
During normal operation the `measurement.time` is null, which then defaults to the current time (not entirely sure if it's the InfluxDB lib or InfluxDB itself creating the timestamp), but a custom strategy could very well manually set a timestamp to the `RuuviMeasurement` and then when the measurement is saved to InfluxDB the timestamp is what was set in the measurement. https://github.com/Scrin/RuuviCollector/blob/2e07aa76a37c826f8a5e184b6035ff3b7d434f7c/src/main/java/fi/tkgwf/ruuvi/utils/InfluxDBConverter.java#L26-L28
The included migrator utility from the old schema to the new schema makes use of this.